src/main/scala/firrtl/passes/AnnotateValidMemConfigs.scala

```diff
@@ -0,0 +1,298 @@
+package firrtl.passes
+
+import firrtl.ir._
+import firrtl._
+import net.jcazevedo.moultingyaml._
+import net.jcazevedo.moultingyaml.DefaultYamlProtocol._
+import AnalysisUtils._
+import scala.collection.mutable.ArrayBuffer
+import firrtl.Mappers._
+
+object CustomYAMLProtocol extends DefaultYamlProtocol {
+  // bottom depends on top
+  implicit val dr = yamlFormat4(DimensionRules)
+  implicit val md = yamlFormat2(MemDimension)
+  implicit val sr = yamlFormat4(SRAMRules)
+  implicit val wm = yamlFormat2(WMaskArg)
+  implicit val sc = yamlFormat11(SRAMCompiler)
+}
+
+case class DimensionRules(
+  min: Int,
+  // step size
+  inc: Int,
+  max: Int,
+  // these values should not be used, regardless of min,inc,max
+  illegal: Option[List[Int]]
```

Comment:
Does Option[List[_]] make sense? Can't it just be a list that is sometimes empty?